hack/dockerfiles/shfmt.Dockerfile

```diff
@@ -0,0 +1,17 @@
+# syntax = docker/dockerfile:1.1-experimental
+FROM mvdan/shfmt:v3.1.2-alpine AS shfmt
+WORKDIR /src
+ARG SHFMT_FLAGS="-i 2 -ci"
```

Comment:
seems inconsistent with moby/moby
https://github.com/moby/moby/blob/a770dc191eea0a88236b4fb5575fe92efd356800/hack/validate/shfmt#L4

Reply:
Yes, but consistent with https://github.com/tonistiigi/fsutil/blob/master/hack/dockerfiles/shfmt.Dockerfile#L4 . This matches https://google.github.io/styleguide/shellguide.html and somewhat similar to the current style these scripts were using.

Reply:
I think I tweaked those at the time to keep the formatting as close as possible to what was there.

Im not at my computer right now; does changing make a big difference?

Reply:
^ FWIW current formatting (at a glance) looks decent, and I'm personally not attached to the exact formatting as is used in Moby. I think what's most important is to keep a consistent (and sensible) format within this project

Reply:
Hmm, looks like I missed that this switches scripts to use spaces for indentation, which really doesn't seem standard (no idea why Google would recommend that)

Reply:
Not sure if we can remain friends after disagreeing with so fundamental question as tabs vs spaces 😄

Reply:
> which really doesn't seem standard

Many projects use this for shell (I'd guess about 50%). Eg. gcc, llvm, rust. Google's styleguide is the closest thing to a standard I know. These scripts previously also used spaces.

https://google.github.io/styleguide/shellguide.html#s5.1-indentation

> Use blank lines between blocks to improve readability. Indentation is two spaces. Whatever you do, don’t use tabs. For existing files, stay faithful to the existing indentation.

Reply:
> Not sure if we can remain friends after disagreeing with so fundamental question as tabs vs spaces 😄

😂😂😂 

> Many projects use this for shell (I'd guess about 50%). Eg. gcc, llvm, rust. Google's styleguide is the closest thing to a standard I know. These scripts previously also used spaces.

I know if would make @tianon sad for starters, but perhaps he could provide his 0.02c (I learned from him that scripts should use tabs, not spaces)

Reply:
Everyone has their own religious reasons and history of why they are in one camp over the other. The only thing that is sure after this has been argued over for countless times is that you can't objectively prove one is better than another for all people.

I have no plans to push any formatting changes similar to this in moby/moby repository or any other project where I haven't written the scripts myself. It just makes sure we are consistent with minimal changes to the previous style and borrowing the exact rules from the set that has been determined by a collection of real-life projects.